src/test/groovy/graphql/parser/IDLParserTest.groovy

```diff
@@ -582,5 +588,162 @@ input Gun {
         (typeDef2.getImplements()[1] as TypeName).getName() == 'Baz'
     }
 
+    def "basic type extensions"() {
+
+        def input = '''
+
+        type Query {
+            bar : String
+        }
+        
+        extend type Query @directiveOnly
+
+        extend type Query @directive {
+            field : String
+        }
+
+        
+        interface Bar {
+            bar : String
+        }
+
+        extend interface Bar @directiveOnly
+        
+        extend interface Bar @directive {
+          iField : String
+        }
+        
+        
+        union FooBar = Foo | Bar
+
+        extend union FooBar @directiveOnly
+
+        extend union FooBar @directive =
+            | Baz 
+            | Buzz
+        
+        
+        enum Numb {
+            A, B, C
+        }
+        
+        extend enum Numb @directiveOnly
+        
+        extend enum Numb @directive {
+            E,F
+        }
+        
+        
+        scalar Scales
+        
+        extend scalar Scales @directiveOnly 
+        
+        extend scalar Scales @directive
+        
+        input Puter {
+            field : String
+        }
+        
+        extend input Puter @directiveOnly
+        
+        extend input Puter @directive {
+            inputField : String
+        }
+            
+        '''
+
+        when:
+        def doc = new Parser().parseDocument(input)
+
+        then:
+
+        // object type extension
+        fromDoc(doc, 0, ObjectTypeDefinition).name == "Query"
+
+        fromDoc(doc, 1, TypeExtensionDefinition).name == "Query"
```

Comment:
I think we should use this PR and rename TypeExtensionDefinition to ObjectTypeExtension to make it in line with the other names

Reply:
ok